lib/Ast.ml

```diff
@@ -1742,28 +1742,9 @@ end = struct
        |Pexp_variant (_, Some _)
        |Pexp_while _ | Pexp_new _ | Pexp_object _ ->
           Some Apply
-      | Pexp_extension
-          ( ext
-          , PStr
-              [ { pstr_desc=
-                    Pstr_eval
-                      ( ( { pexp_desc=
-                              ( Pexp_new _ | Pexp_object _ | Pexp_while _
-                              | Pexp_for _ | Pexp_function _ | Pexp_fun _
-                              | Pexp_try _ | Pexp_match _ | Pexp_let _ )
-                          ; _ } as e )
-                      , _ )
-                ; _ } ] )
-        when Source.extension_using_sugar ~name:ext ~payload:e.pexp_loc ->
-          Some Apply
-      | Pexp_extension
-          ( ext
-          , PStr
-              [ { pstr_desc=
-                    Pstr_eval (({pexp_desc= Pexp_sequence _; _} as e), _)
-                ; _ } ] )
+      | Pexp_extension (ext, PStr [{pstr_desc= Pstr_eval (e, _); _}])
```

Comment:
can you explain this change ?

Reply:
Because in the case where we use the sugared syntax this function should return the same for `keyword%ext` and `keyword`